mutiny-wasm/src/indexed_db.rs

```diff
@@ -0,0 +1,498 @@
+                    indexed_db
+                        .transaction(&[WALLET_OBJECT_STORE_NAME], TransactionMode::ReadWrite)
+                        .map_err(|e| {
+                            MutinyError::read_err(
+                                anyhow!("Failed to create indexed db transaction: {e}").into(),
+                            )
+                        })
+                } else {
+                    Err(MutinyError::read_err(MutinyStorageError::IndexedDBError))
+                }
+            })?;
+
+        let store = tx.store(WALLET_OBJECT_STORE_NAME).map_err(|e| {
+            MutinyError::read_err(anyhow!("Failed to create indexed db store {e}").into())
+        })?;
+
+        // delete from indexed db
+        store
+            .delete(&JsValue::from(key))
+            .await
+            .map_err(|_| MutinyError::write_err(MutinyStorageError::IndexedDBError))?;
+
+        tx.done()
+            .await
+            .map_err(|_| MutinyError::write_err(MutinyStorageError::IndexedDBError))?;
+
+        Ok(())
+    }
+
+    pub(crate) async fn read_all(
+        indexed_db: &Arc<RwLock<Option<Rexie>>>,
+    ) -> Result<HashMap<String, Value>, MutinyError> {
+        let store = {
+            let tx = indexed_db
+                .try_read()
+                .map_err(|e| MutinyError::read_err(e.into()))
+                .and_then(|indexed_db_lock| {
+                    if let Some(indexed_db) = &*indexed_db_lock {
+                        indexed_db
+                            .transaction(&[WALLET_OBJECT_STORE_NAME], TransactionMode::ReadOnly)
+                            .map_err(|e| {
+                                MutinyError::read_err(
+                                    anyhow!("Failed to create indexed db transaction: {e}").into(),
+                                )
+                            })
+                    } else {
+                        Err(MutinyError::read_err(MutinyStorageError::IndexedDBError))
+                    }
+                })?;
+            tx.store(WALLET_OBJECT_STORE_NAME).map_err(|e| {
+                MutinyError::read_err(anyhow!("Failed to create indexed db store {e}").into())
+            })?
+        };
+
+        let mut map = HashMap::new();
+        let all_json = store.get_all(None, None, None, None).await.map_err(|e| {
+            MutinyError::read_err(anyhow!("Failed to get all from store: {e}").into())
+        })?;
+        for (key, value) in all_json {
+            let key = key
+                .as_string()
+                .ok_or(MutinyError::read_err(MutinyStorageError::Other(anyhow!(
+                    "key from indexedDB is not a string"
+                ))))?;
+
+            let json: Value = value.into_serde()?;
+            map.insert(key, json);
+        }
+
+        Ok(map)
+    }
+
+    async fn build_indexed_db_database() -> Result<Rexie, MutinyError> {
+        let rexie = Rexie::builder(WALLET_DATABASE_NAME)
```

Comment:
So now everything will be saved into `WALLET_DATABASE_NAME` but as a key value store? So the gossip, logging, and peer storage stuff will go away? Kind of a breaking change except those components were not critical right? 

Do all the keys in the wallet DB map exactly the same with this PR too? Trying to figure out if all clients will need a fresh instance after this upgrade.

Reply:
> So the gossip, logging, and peer storage stuff will go away? Kind of a breaking change except those components were not critical right?

yeah pretty much

Reply:
Seems so. We just have to reconnect peer. I'll make sure to test that.